### ISSUE TITLE
JS conflict with Fabric

### ISSUE THREAD

**Issue by danyj** [YJSGframework/yjsg#32]
ref 
http://www.youjoomla.com/joomla_support/showthread.php?p=64546#post64546

**Comment by danyj** [comment]
@Hugh  if you are referencing this 
https://github.com/YJSGframework/yjsg/blob/master/assets/src/yjsg.site.plugins.js#L58

please try to do a workaround that would fit us both and we will gladly commit to the fix.

**Comment by danyj** [comment]
Or see if this one work better 
https://github.com/carhartl/jquery-cookie/blob/master/src/jquery.cookie.js

**Comment by danyj** [comment]
OK I think I found the issue https://github.com/carhartl/jquery-cookie/issues/340

**Comment by hugh** [comment]
(just FYI- probably you mean a different hugh)

**Comment by danyj** [comment]
sorry bud , yes , waiting on Hugh from Fabrik

**Comment by danyj** [comment]
I think this one should work , I dont have Fabric so cant test ,  
https://github.com/js-cookie/js-cookie/blob/master/src/js.cookie.js

use that plugin instead of   this one https://github.com/YJSGframework/yjsg/blob/master/assets/src/yjsg.site.plugins.js#L51-L58

if it works we will release and update

**Comment by skurvish** [comment]
danyj, I can try it on my site. I am not sure what you want me to change though.

**Comment by danyj** [comment]
@skurvish   

replace this  
https://github.com/YJSGframework/yjsg/blob/master/assets/src/yjsg.site.plugins.js#L51-L58

with this 
https://github.com/js-cookie/js-cookie/blob/master/src/js.cookie.js

and see if you have issues

**Comment by danyj** [comment]
or yo can PM Neo in forum with site admin info and he will try to test

**Comment by skurvish** [comment]
the plugins/system/yjsg/assets/src/yjsg.site.plugins.js file I have is minified. Do you just want me to cut and paste the contents of the js.cookie.js file into it?

**Comment by danyj** [comment]
yes please , just make sure to remove this one first 

https://github.com/YJSGframework/yjsg/blob/master/assets/src/yjsg.site.plugins.js#L51-L58

**Comment by danyj** [comment]
One more while you at it 

replace 
https://github.com/YJSGframework/yjsg/blob/e43be49543e6b15ca537373ed1e14f69d003299d/assets/src/yjsg.site.js#L438
with 

      if (Cookies.get('ieNotice')) return;


and 

https://github.com/YJSGframework/yjsg/blob/e43be49543e6b15ca537373ed1e14f69d003299d/assets/src/yjsg.site.js#L454

with 

    Cookies.set('ieNotice', true);

**Comment by skurvish** [comment]
Made all 3 changes, problem still exists.

**Comment by danyj** [comment]
ok , we need to get Hugh in to advise what define he was talking about. See if you can get him to join us please. Thnx for testing and dont worry , we will find a fix.

**Comment by cheesegrits** [comment]
Hey.  I'm @cheesegrits on github, not hugh.  :)

I'm still trying to figure this one out.  I'm really not sure what the "mismatch" is, and unfortunately the error message is spectacularly unhelpful.  I had thought it was the use of the ['jquery'] define, but ... :/

I'll try and find some time this week to work on it.

**Comment by danyj** [comment]
@cheesegrits (lol, nice nick ) hey bud , thnx for drooping in. Listen , try to kill  our js  plugins one by one and see if that helps at all. I still think is just a script order conflict but not sure.

**Comment by skurvish** [comment]
Hugh has been swamped. As mentioned in another thread I have had to migrate away fro YJSG to another framework as a result of this incompatibility. It is too bad because I really like the framework and the effort to migrate was significant, but it is what it is.

**Comment by danyj** [comment]
ok , il close the issue for now and if @cheesegrits  gets to it we can reopen since this is a low priority right now.

@skurvish  dropping any framework becuase of one JS error is a huge step but good luck.

**Comment by skurvish** [comment]
Well if the js error causes a vital function on the website to fail, and there is no fix in the foreseeable future, then I must drop the framework. It wish it wasn’t so but it is.

**Comment by danyj** [comment]
the issue that we still think is valid was completely explained here 
http://www.youjoomla.com/joomla_support/showthread.php?t=14198#edit64512

and YJSG is not doing that but Fabric is.

**Comment by skurvish** [comment]
Maybe, but it isn’t getting fixed any time soon no matter who created the problem.

**Comment by cheesegrits** [comment]
Unfortunately I just don't have time to pursue this.  It's just the nature of the shared sandbox beast, with extensions having to play nice with multiple template frameworks, using conflicting JS module loading techniques.  Hopefully one day J! will standardize on dependency loading.
